development/image-url-helper/pkg/missing/missing.go

```diff
@@ -0,0 +1,55 @@
+package missing
+
+import (
+	"strings"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/kyma-project/test-infra/development/image-url-helper/pkg/common"
+)
+
+// CheckForMissingImages checks if images exist and returns ComponentImageMap of nonexistent images
+func CheckForMissingImages(allImages common.ComponentImageMap, missingImages common.ComponentImageMap) error {
+
+	for imageURL, image := range allImages {
+		missing, err := isImageMissing(image)
+		if err != nil {
+			return err
+		}
+
+		if missing {
+			if !strings.Contains(err.Error(), "Failed to fetch") {
```

Comment:
err variable will be always nil and this will throw exception. If err will be not nil function will return in line 17 because you check the same err variable.